includes/class-wc-subscriptions-email-notifications.php

```diff
@@ -0,0 +1,297 @@
+				}
+				break;
+			case 'woocommerce_scheduled_subscription_customer_notification_trial_expiration':
+				$notification = $emails['WCS_Email_Customer_Notification_Free_Trial_Expiration'];
+				break;
+			case 'woocommerce_scheduled_subscription_customer_notification_expiration':
+				$notification = $emails['WCS_Email_Customer_Notification_Subscription_Expiration'];
+				break;
+		}
+
+		if ( $notification ) {
+			$notification->trigger( $subscription_id );
+		}
+	}
+
+	public static function notifications_globally_enabled() {
+		return ( 'yes' === get_option( WC_Subscriptions_Admin::$option_prefix . self::$switch_setting_string )
+				&& get_option( WC_Subscriptions_Admin::$option_prefix . self::$offset_setting_string ) );
+	}
+
+	/**
+	 * Should the emails be sent out?
+	 *
+	 * @return bool
+	 */
+	public static function should_send_notification() {
+		$notification_enabled = true;
+
+		if ( WCS_Staging::is_duplicate_site() ) {
+			$notification_enabled = false;
+		}
+
+		$allowed_env_types = [
+			'production',
+		];
+		if ( ! in_array( wp_get_environment_type(), $allowed_env_types, true ) ) {
+			$notification_enabled = false;
+		}
+
+		// If Customer notifications are disabled in the settings by a global switch, or there is no offset set, don't send notifications.
+		if ( ! self::notifications_globally_enabled() ) {
+			$notification_enabled = false;
+		}
+
+		/**
+		 * Enables/disables all customer subscription notifications.
+		 *
+		 * Values 'yes' or 'no' expected, since it works with WC_Settings_API.
+		 *
+		 * @since 8.0.0
+		 *
+		 * @param string $notification_enabled
+		 */
+		return apply_filters( 'wcs_customer_email_notifications_enabled', $notification_enabled );
+	}
+
+	/**
+	 * @param $subscription
+	 *
+	 * @return bool
+	 */
+	public static function subscription_period_too_short( $subscription ) {
+		$period   = $subscription->get_billing_period();
+		$interval = $subscription->get_billing_interval();
+
+		// By default, there are no shorter periods than days in WCS, so we ignore hours, minutes, etc.
+		if ( $period <= 2 && 'day' === $interval ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Adds actions to the admin edit subscriptions page.
+	 *
+	 * @param array $actions An array of available actions
+	 * @return array An array of updated actions
+	 */
+	public static function add_notification_actions( $actions ) {
+		global $theorder;
+
+		if ( wcs_is_subscription( $theorder ) ) {
+			$subscription = $theorder;
+			//TODO: confirm if these statuses make sense.
+			$allowed_statuses = [
+				'active',
+				'on-hold',
+				'pending-cancellation',
+			];
+
+			if ( ! in_array( $subscription->get_status(), $allowed_statuses, true ) ) {
+				return $actions;
+			}
+
+			if ( $subscription->get_date( 'trial_end' ) ) {
+				$actions['wcs_customer_notification_free_trial_expiration'] = esc_html__( 'Send Free Trial Expiration notification', 'woocommerce-subscriptions' );
```

Comment:
Super small nit-pick comment incoming.
The capitalization of these order notes isn't consistent with other similar WooCommerce's order actions:

| Order | Subscription |
|--------|--------|
| ![VpmypC.png](https://github.com/user-attachments/assets/7c3e8c74-48df-4f9a-986d-db955e09cbf4) | ![image](https://github.com/user-attachments/assets/eed095a9-4dcd-487a-9fea-4c6b3b47c058) | 

While I agree with your capitalization, I'm thinking we should change it to be consistent :)

```suggestion
				$actions['wcs_customer_notification_free_trial_expiration'] = esc_html__( 'Send free trial expiration notification', 'woocommerce-subscriptions' );
```

Reply:
Makes sense. Fixed in [868ff38](https://github.com/Automattic/woocommerce-subscriptions-core/pull/648/commits/868ff381e18aeea1480d61c61723932e16e43083)